x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js

```diff
@@ -411,20 +410,8 @@ export class VectorStyle extends AbstractStyle {
   }
 
   renderLegendDetails() {
-    const loadRows = async () => {
-      const styles = await this._getLegendDetailStyleProperties();
-      const promises = styles.map(async style => {
-        return {
-          label: await style.getField().getLabel(),
-          fieldFormatter: this._getFieldFormatter(style.getField().getName()),
-          meta: this._getFieldMeta(style.getField().getName()),
-          style,
-        };
-      });
-      return await Promise.all(promises);
-    };
-
-    return <VectorStyleLegend loadRows={loadRows} />;
+    const stylesPromise = this._getLegendDetailStyleProperties();
```

Comment:
I think its pretty weird to pass promises instead of the function. Why not just pass `getLegendDetailStyleProperties={this._getLegendDetailStyleProperties}` into `VectorStyleLegend `?